[#57704] Resolve pending spec in `direct_ifc_upload_spec.rb`

akabiru opened the pull request:
The JS form validity was previously flaky but seems resolved now. I can't point out exactly what has changed.

Cheers to @cbliard who caught that it was now passing

https://community.openproject.org/work_packages/57704

----

```bash
❯ docker compose exec backend-test script/bulk_run_rspec modules/bim/spec/features/ifc_models/direct_ifc_upload_spec.rb
1 tests to run
logs: /home/dev/openproject/tmp/bulk_run/logs
================================================================================
Running modules/bim/spec/features/ifc_models/direct_ifc_upload_spec.rb 5 times
================================================================================
------------------------------------ Run 0 -------------------------------------
ok
------------------------------------ Run 1 -------------------------------------
ok
------------------------------------ Run 2 -------------------------------------
ok
------------------------------------ Run 3 -------------------------------------
ok
------------------------------------ Run 4 -------------------------------------
ok
PASSED
Saving
Done
results in /home/dev/openproject/tmp/bulk_run
```

akabiru committed:
[#57704] Resolve pending spec in direct_ifc_upload_spec.rb

The JS form validity was previously flaky but seems resolved now. I can't point out exactly what has changed.

https://community.openproject.org/work_packages/57704

akabiru commented:
Welp, looks like it's still flaky @cbliard 😅 👀  perhaps an ordering issue- I'll attempt your tip on rspec bisect 

<details> <summary>bisect result</summary>

```ruby
Bisect complete! Reduced necessary non-failing examples from 8 to 2 in 36 minutes 52 seconds.

The minimal reproduction command is:
  rspec './modules/bim/spec/features/ifc_models/direct_ifc_upload_spec.rb[1:1:2:1:1]' './modules/calendar/spec/features/calendar_sharing_spec.rb[1:1:1:1,1:2:1:1,1:2:2:1,1:2:2:2,1:2:2:3,1:2:2:4,1:2:2:5,1:2:2:6,1:3:1]' './modules/github_integration/spec/features/work_package_github_tab_spec.rb[1:1:1:1,1:1:1:2:1,1:1:1:3:1,1:1:1:4:1,1:2:1:1,1:2:1:2:1,1:2:1:3:1,1:2:1:4:1,1:3:1:1,1:3:1:2:1,1:3:1:3:1,1:3:1:4:1]' './modules/meeting/spec/features/meetings_attachments_spec.rb[1:1:1:1,1:2:1]' './modules/my_page/spec/features/my/work_package_watcher_widget_spec.rb[1:1]' './modules/overviews/spec/features/low_permissions_page_creation_spec.rb[1:1]' './modules/reporting/spec/features/calculations_spec.rb[1:1]' './modules/reporting/spec/features/update_cost_report_spec.rb[1:1]' './modules/team_planner/spec/features/onboarding/team_planner_onboarding_tour_spec.rb[1:1:1]' './modules/team_planner/spec/features/team_planner_dates_spec.rb[1:1:1]' './modules/team_planner/spec/features/team_planner_global_create_spec.rb[1:1:1:1,1:2:1:1,1:2:2:1:1,1:2:2:2:1,1:2:3:1:1]' './spec/features/menu_items/wiki_menu_item_spec.rb[1:1:1,1:2]' './spec/features/users/create_spec.rb[1:1:1:1,1:1:1:2,1:1:1:3:1,1:1:1:3:2,1:2:1:1,1:2:1:2,1:2:1:3:1,1:2:1:3:2,1:3:1:1:1,1:3:1:1:2,1:3:1:1:3:1,1:3:1:1:3:2,1:4:1:1:1,1:4:1:1:2]' './spec/features/work_packages/details/markdown/activity_comments_spec.rb[1:1:1:1:1,1:1:1:1:2,1:1:1:1:3,1:1:1:2:1,1:1:1:3:1:1,1:1:1:3:2:1:1:1:1,1:1:1:3:2:1:2:1:1,1:1:1:4:1,1:1:2:1,1:1:3:1,1:1:4,1:2:1]' './spec/features/work_packages/table/hierarchy/hierarchy_sorting_spec.rb[1:1]' './spec/features/work_packages/tabs/activity_revisions_spec.rb[1:1:1:1:1:1:1,1:1:1:1:2:1:1,1:1:1:1:3,1:1:1:1:4,1:1:1:1:5,1:1:1:2:1:1:1,1:1:1:2:2,1:2:1:1:1:1:1,1:2:1:1:2:1:1,1:2:1:1:3,1:2:1:1:4,1:2:1:1:5,1:2:1:2:1:1:1,1:2:1:2:2]' --seed 32482
```

</details>

cbliard commented:
Thanks @akabiru . Indeed, it is flaky. The bisect did not seem to work as a lot of tests appeared to be failing when you ran it. I too noticed that `./spec/features/types/form_configuration_query_spec.rb` and `./modules/bim/spec/features/model_viewer_spec.rb` are failing locally on my machine. But the `modules/bim/spec/features/ifc_models/direct_ifc_upload_spec.rb` one did not fail. 

This is frustrating.

cbliard committed:
Fix direct_ifc_upload_spec test

When the file input gets a file, it makes a request to the backend to
check for the file size. This request takes some time. The test was not
waiting for the request to complete before checking the form validity.

By adding a check on the validation message of the file input, we make
sure the test waits for the request to complete and the that form
validity check is correct.

akabiru commented on modules/bim/spec/features/ifc_models/direct_ifc_upload_spec.rb:
```diff
@@ -36,12 +36,13 @@
 
         page.attach_file("file", ifc_fixture.path, visible: :all)
 
+        expected_validation_message = I18n.t("activerecord.errors.messages.file_too_large", count: 1024)
+        expect(page).to have_field(type: "file", validation_message: expected_validation_message)
+
```
Nice one @cbliard 🙌🏾 🤩
